client/src/components/CheckInCal.jsx

```diff
@@ -21,7 +30,69 @@ class CheckInCal extends React.Component {
     return current;
   }
 
-  getSundays(y, n) {
+  createMonth(today, input_month) {
+    let days = [];
+    if (input_month === 2) {
+      days = Array.from({length: 28}, (_, i) => i + 1);
+    }
+    let thirty = [4, 6, 9, 11];
+    if (thirty.indexOf(input_month) === -1) {
+      days = Array.from({length: 31}, (_, i) => i + 1);
+    } else {
+      days = Array.from({length: 30}, (_, i) => i + 1);
+    }
+    const length = days.length;
+    const sundays = this.getSundays(2021);
+    let prev_sundays_inventory = [];
+    let next_sundays_inventory = [];
+    sundays.forEach(month => {
+      if (month[0] === (input_month - 1)) {
+        for (let i = 0; i < month.length; i++) {
+          if (i === 0) {
+            if (month[i] === days[0]) {
+              console.log('current month begins on a Sunday');
+              if (length === 31) {
+                days = days.concat([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11])
+              } else if (length === 31) {
+                days = days.concat([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12])
+              } else {
+                days = days.concat([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14])
+              }
+              break;
+            }
+          }
+        }
+      }
+      else if (month[0] === (input_month - 2)) {
+        prev_sundays_inventory.push(month[1]);
+      } else if (month[0] === (input_month)) {
+        next_sundays_inventory.push(month[1]);
+      }
+    });
+    let last_previous_sunday = prev_sundays_inventory.pop();
+    let previous_month_length = 0;
+    if (length === 31) {
+      previous_month_length = 30;
+    } else if (length === 30 || length === 28) {
+      previous_month_length = 31;
+    }
+    let preLoad = [last_previous_sunday];
+    let diff = (previous_month_length - last_previous_sunday);
+    while (diff > 0) {
+      let numb = last_previous_sunday += 1;
+      preLoad.push(numb);
+      diff --;
+    }
+    preLoad = preLoad.concat(days);
+    const days_needed = (42 - preLoad.length);
+    let next_days = [];
+    next_days = Array.from({length: days_needed}, (_, i) => i + 1);
+    preLoad = preLoad.concat(next_days);
+    console.log(preLoad);
+    return preLoad;
+  }
+
+  getSundays(y) {
```

Comment:
I Alluded to this in a prev comment, but major kudos. you should be v. v. v. proud! I'm super impressed.

Reply:
Ty!!! It was gnarly but ultimately like a semi-intense toy prob.